src/gui/maptools/qgsmaptoolcapture.cpp

```diff
@@ -343,12 +342,13 @@ bool QgsMapToolCapture::tracingAddVertex( const QgsPointXY &point )
         mCaptureCurve = *qgsgeometry_cast<QgsCompoundCurve *>( curved.constGet() );
       }
     }
-  }
 
-  // sync the snapping matches list
-  const int pointAfter = mCaptureCurve.numPoints();
-  for ( ; pointBefore < pointAfter; ++pointBefore )
-    mSnappingMatches.append( QgsPointLocator::Match() );
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+    mSnappingMatches = QVector( mCaptureCurve.numPoints(), QgsPointLocator::Match() ).toList();
```

Comment:
This doesn't look right, we should be keeping the existing entries but resizing

Reply:
done 54803792f9b